sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -381,7 +383,9 @@ public VoidResponse downloadWithResponse(OutputStream stream, BlobRange range, R
             .flatMapMany(res -> res.getValue()
                 .doOnNext(bf -> {
                     try {
-                        stream.write(bf.array());
+                        WritableByteChannel channel = Channels.newChannel(stream);
```

Comment:
Should this channel instantiation be outside of the reactive stream? This is creating a new channel each time another item is emitted from the stream.